### PR TITLE
Better documentation and termination state handling in Tactic FSMs

### PR DESCRIPTION
**Pull request by jonathanlew**
### Description
* Improved documentation in Tactic FSMs, especially with the function definitions
* Fixed situations where an FSM in the terminal state wouldn't transition out when the conditions aren't true
* Rearranged the folders for tactics that use fsms

### Testing Done
added unit tests

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/ai/hl/stp/tactic/kick/kick_fsm.h [hunk 1]
```diff
@@ -45,6 +56,13 @@ struct KickFSM
                 processEvent(GetBehindBallFSM::Update(control_params, event.common));
             };
 
+        /**
+         * Guard that checks if the ball has been chicked
```
Comment (tarakong01) [inline src/software/ai/hl/stp/tactic/kick/kick_fsm.h:60]: is the word "chicked" just the combination of "chipped" and "kicked"?

File: src/software/ai/hl/stp/tactic/kick/kick_fsm_test.cpp [hunk 2]
```diff
@@ -13,27 +13,32 @@ TEST(KickFSMTest, test_transitions)
                                           .kick_speed_meters_per_second = 1.2};
 
     HFSM<KickFSM> fsm;
-    // Start in GetBehindBallFSM state's idle_state
+
+    // Start in GetBehindBallFSM state's get_behind_ball_state
     EXPECT_TRUE(fsm.is(boost::sml::state<GetBehindBallFSM>));
     EXPECT_TRUE(fsm.is<decltype(boost::sml::state<GetBehindBallFSM>)>(
```
Comment (tarakong01) [inline src/software/ai/hl/stp/tactic/kick/kick_fsm_test.cpp:19]: What's the difference between the two EXPECT_TRUE statements?

**Review by tarakong01 — COMMENTED** [review]
Left a couple of questions :)

File: src/software/ai/hl/stp/tactic/kick/kick_fsm.h (hunk 1 above)
Comment (jonathanlew) [reply src/software/ai/hl/stp/tactic/kick/kick_fsm.h:60]: yes

File: src/software/ai/hl/stp/tactic/kick/kick_fsm_test.cpp (hunk 2 above)
Comment (jonathanlew) [reply src/software/ai/hl/stp/tactic/kick/kick_fsm_test.cpp:19]: the first line checks that the fsm is in _any_ substate of GetBehindBallFSM and the second line checks that it's in a particular substate of the FSM.

File: src/software/ai/hl/stp/tactic/move/move_fsm.h [hunk 3]
```diff
@@ -29,23 +28,32 @@ struct MoveFSM
...
+        // move_s is the _state_ used in the transition table
         const auto move_s = state<move_state>;
 
         // update_e is the _event_ that the MoveFSM responds to
         const auto update_e = event<Update>;
 
-        // this action sets the intent to a move intent corresponding to the update_e
-        // event
+        /**
+         * This is an Action sets the intent to a move intent corresponding to the
```
Comment (itsarune) [inline src/software/ai/hl/stp/tactic/move/move_fsm.h]: This is an Action *that* sets the intent...

**Review by itsarune — COMMENTED** [review]
just spelling nits

**Review by akhilveeraghanta — DISMISSED** [review]
lgtm!, pending existing spelling comment

File: src/software/ai/hl/stp/tactic/move/move_fsm.h (hunk 3 above)
Comment (jonathanlew) [reply src/software/ai/hl/stp/tactic/move/move_fsm.h]: fixed

**Review by akhilveeraghanta — APPROVED** [review]

**Review by itsarune — APPROVED** [review]
looks good to me

**Pull request merged**